Lib/bdb.py

```diff
@@ -439,6 +439,14 @@ def set_break(self, filename, lineno, temporary=False, cond=None,
             return 'Line %s:%d does not exist' % (filename, lineno)
         self._add_to_breaks(filename, lineno)
         bp = Breakpoint(filename, lineno, temporary, cond, funcname)
+        # After we set a new breakpoint, we need to search through all frames
+        # and set f_trace to trace_dispatch if there could be a breakpoint in
+        # that frame.
```

Comment:
I don't understand this comment. If there is a breakpoint in that frame, would it not have f_trace set already?

Reply:
No, that's what this patch fixed. For existing frames, there could be frame that does not have `f_trace` set. And setting a new breakpoint (`b func`) on that frame did not set `f_trace` before this patch, which made the program fail to hit the breakpoint.